ambry-store/src/main/java/com.github.ambry.store/Journal.java

```diff
@@ -186,4 +187,26 @@ Long getCrcOfKey(StoreKey key) {
   StoreKey getKeyAtOffset(Offset offset) {
     return journal.get(offset);
   }
+
+  /**
+   * Puts the {@link Journal} into bootstrap mode to ignore the {@code maxEntriesToJournal} constraint temporarily.
+   */
+  void startBootstrap() {
```

Comment:
The original idea was to limit this bootstrap mode to fix what is broken. Bootstrap mode is turned off by default to mimic the old behavior since Journal is also created and used in other places (such as compaction).